minix/commands/autopart/autopart.c

```diff
@@ -1531,8 +1531,13 @@ void m_read(int ev, int *biosdrive)
 	stat_start(0);
 	fflush(stdout);
 
-	if ((device= open(curdev->name, mode= O_RDWR, 0666)) < 0) {
-		if (device >= 0) { close(device); device= -1; }
+	device = open(curdev->name, O_RDWR, 0666);
+	if (device < 0) {
+		return;
+	}
+	else if (device >= 0) {
+		close(device);
+		device = -1;
```

Comment:
This change is not correct, as it inverts the behavior of the code that was there in the first place. The original line 1535 can only trigger if device is set to a less-than-zero value, so all that needs to be done is removal of that particular line.